dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/DatabaseStore.java

```diff
@@ -10,24 +10,53 @@
  *******************************************************************************/
 package com.ibm.wsspi.persistence;
 
+import java.util.Map;
+
 /**
  * Enables configuration of a database as a persistent store that can be shared by multiple components.
  */
 public interface DatabaseStore {
+
     /**
      * Create a persistence service unit for the specified entity classes.
      * The invoker of this method is responsible for closing the persistence service unit
      * and for participating in DDL generation.
-     * 
+     *
      * The persistence service feature is currently not supported for informix.
-     * 
+     *
      * @param loader class loader for the entity classes.
      * @param entityClasses list of entity classes.
      * @return the persistence service unit.
      * @throws Exception if a failure occurs.
      */
     PersistenceServiceUnit createPersistenceServiceUnit(ClassLoader loader, String... entityClassNames) throws Exception;
 
+    /**
+     * Create a persistence service unit for the specified entity classes.
+     * Allows the invoker to pass configuration properties.
+     * The invoker of this method is responsible for closing the persistence service unit
+     * and for participating in DDL generation.
+     *
+     * The persistence service feature is currently not supported for informix.
+     *
+     * @param loader class loader for the entity classes.
+     * @param properties map of configuration properties
+     * @param entityClasses list of entity classes.
+     * @return the persistence service unit.
+     * @throws Exception if a failure occurs.
+     */
+    PersistenceServiceUnit createPersistenceServiceUnit(ClassLoader loader, Map properties, String... entityClassNames) throws Exception;
+
+    /**
+     * Obtain the database product name from the database to which the data source connects.
+     * <p>
+     * {@link com.ibm.ws.jdbc.WSDataSource#getDatabaseProductName()}
+     * {@link java.sql.DatabaseMetaData#getDatabaseProductName()}
+     *
+     * @return the database product name or 'UNKNOWN' if unable
+     */
+    public String getDatabaseProductName();
```

Comment:
Rather than going to all of the trouble of getting the db product name and interpreting the meaning of the value, you could just ask the DatabaseMetaData directly if it supports the isolation level,
https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#supportsTransactionIsolationLevel(int)

Reply:
The persistence service would first create a DataSource with `Connection.TRANSACTION_READ_COMMITTED`, so that we can get a connection and therefore the `DatabaseMetaData`, so that we can ask if it supports `Connection.TRANSACTION_READ_UNCOMMITTED`? 

Getting access to the javax.sql.DatabaseMetaData doesn't seem quicker and makes the persistence service care more about this decision that they aren't responsible for in the first place.

Reply:
WSDataSource doesn't actually know anything. I'm pretty sure it would need to get the DatabaseMetaData and ask it